### DISCUSSION TITLE
Feature Gates/Flags

### DISCUSSION THREAD

**Discussion by perdasilva** [operator-framework/rukpak#344] [General]
I think we should start thinking about introducing feature flags or at least have a plan for how and when. Especially how we want to deal with API changes in this context.

This will help us get feedback on ideas without disruption and with downstreaming RukPak

**Comment by timflannagan** [comment]
It seems like there's two potential classes of feature gates:

- Feature gates with the core rukpak APIs (e.g. configuring something like DefaultStorageClass for an admission controller)
- Feature gates with the individual provisioner implementations

Do we have any idea on whether individual StorageClass provisioners store feature gates in the StorageClass' set of parameters, and whether it makes sense to store feature gates in the ProvisionerClass API? I'd guess that pattern doesn't exist, or doesn't make much sense, and so we may need to use feature gates on the individual provisioner executables?
